chain-tx-enclave-next/tx-query-next/app-runner/src/main.rs

```diff
@@ -0,0 +1,112 @@
+mod zmq_helper;
+
+use std::{future::Future, io, pin::Pin};
+
+use aesm_client::AesmClient;
+use enclave_runner::{
+    usercalls::{AsyncListener, AsyncStream, UsercallExtension},
+    EnclaveBuilder,
+};
+use sgxs_loaders::isgx::Device;
+use structopt::StructOpt;
+use tokio::net::{TcpListener, TcpStream};
+
+use zmq_helper::ZmqHelper;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "tx-query-runner", about = "tx-query enclave runner")]
+struct Options {
+    /// Path to enclave file (`.sgxs`). Note that `.sig` file is also expected to be at same
+    /// location
+    #[structopt(short, long)]
+    pub enclave_path: String,
+    /// ZeroMQ connection string of tx-validation server (now in chain-abci). E.g.
+    /// `ipc://enclave.ipc` or `tcp://127.0.0.1:25933`
+    #[structopt(short, long)]
+    pub zmq_conn_str: String,
+    /// `ra-sp-server` address for remote attestation. E.g. `0.0.0.0:8989`
+    #[structopt(short, long)]
+    pub sp_address: String,
+    /// tx-query server address. E.g. `127.0.0.1:3443`
+    #[structopt(short, long)]
+    pub address: String,
+    /// Use dummy signature in testing
+    #[structopt(short, long)]
+    pub test: bool,
+}
+
+#[allow(clippy::type_complexity)]
+impl UsercallExtension for Options {
+    fn connect_stream<'future>(
+        &'future self,
+        addr: &'future str,
+        _local_addr: Option<&'future mut String>,
+        _peer_addr: Option<&'future mut String>,
+    ) -> Pin<Box<dyn Future<Output = io::Result<Option<Box<dyn AsyncStream>>>> + 'future>> {
```

Comment:
the desugared type is still needed even after the upgrade?

Reply:
Yes. `async fn`s are not allowed in traits.